src/Microsoft.DotNet.XHarness.CLI/Commands/Android/AndroidInstallCommand.cs

```diff
@@ -71,7 +71,7 @@ protected override ExitCode InvokeCommand(ILogger logger)
         ILogger logger,
         string apkPackageName,
         string appPackagePath,
-        IEnumerable<string> apkRequiredArchitecture,
+        IReadOnlyCollection<string> requiredArchitecture,
```

Comment:
Shouldn't it be called `requiredArchitectures`?

In general I found the naming a bit confusing. There is `requiredArchitecture`, `apkRequiredArchitecture` and `apkSupportedArchitecture` which all seem similar. If I understand the logic correctly I would propose to rename them to something like:
- `requiredArchitecture` -> `requestedArchitectures`
- `apkRequiredArchitecture` -> `selectedArchitectures` / `actualArchitectures` or something like this
- `apkSupportedArchitectures` - this one is ok, `supportedArchitectures` would be fine as well

You can also extract the part where you determine `apkRequiredArchitecture` to separate method which would hide one of those variables and simplify that part to just (using proposed naming):
```
var selectedArchitectures = SelectAvailableArchitectures(requestedArchitectures, appPackahePath);
```

Reply:
Yeah, this naming was confusing from the start. You're right, I can change it probably a bit more since I am tending to this already

Reply:
I didn't end up putting it into a method because I also need to return the exit code in case of a failure (and also pass in the logger) so it was becoming too much of a hassle